### ISSUE TITLE
Reverse or Modified Order Optimization 

### ISSUE THREAD

**Issue by Jashby1997** [CoinAlpha/hummingbot#1757]
**Feature Suggestion**
This feature will use the idea of filling the gap like the optimization feature in place but in reverse as i'm not using the live market price. 

For example: If i have a buy price of a coin targeted at £9 and its currently being sold at £8 when the order to buy the coin at £9 is place it isn't filled, and instead an error comes up as this is a market buy. 

Would there be a way to get the bot to buy the price up from £8 to £9 (assuming i had the funds)

**Impact**

This will allow me to continue to place buy orders in sudden drops caused by volatility, if the price drops suddenly and falls out of my buy range (Lower then it) orders wont get canceled

**Additional context**
Basically a feature that allows orders to be placed like normal but if the spread set is exceeded then i need a feature turn these normal orders to turn into market orders
